From 5cf345692c9b9ce89e0b521c0b4b220c801dfb59 Mon Sep 17 00:00:00 2001
From: Benjamin Gilbert <bgilbert@redhat.com>
Date: Tue, 9 Jul 2019 19:53:49 -0400
Subject: [PATCH 4/4] ovl: drop test for upperdir used as lowerdir

This is accidentally triggered by multiple versions of Docker.  Drop
the test pending a Docker fix.

Reverts part of 0319ef1d40ff39d2c0f942a46fb73918669b2350.

https://github.com/moby/moby/issues/39475
---
 fs/overlayfs/super.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/fs/overlayfs/super.c b/fs/overlayfs/super.c
index 2d028c02621f..64bca1f53884 100644
--- a/fs/overlayfs/super.c
+++ b/fs/overlayfs/super.c
@@ -1289,12 +1289,6 @@ static int ovl_get_lower_layers(struct super_block *sb, struct ovl_fs *ofs,
 		if (err < 0)
 			goto out;
 
-		err = -EBUSY;
-		if (ovl_is_inuse(stack[i].dentry)) {
-			pr_err("overlayfs: lowerdir is in-use as upperdir/workdir\n");
-			goto out;
-		}
-
 		err = ovl_setup_trap(sb, stack[i].dentry, &trap, "lowerdir");
 		if (err)
 			goto out;
-- 
2.21.0

